unit-tests/live/options/test-options-watcher.py

```diff
@@ -0,0 +1,52 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+# test:device each(D400*)
+
+import pyrealsense2 as rs
+from rspy import test
+from rspy import log
+import time
+
+dev = test.find_first_device_or_exit()
+depth_sensor = dev.first_depth_sensor()
+product_line = dev.get_info(rs.camera_info.product_line)
+
+changed_options = 0
+
+def notification_callback( opt_list ):
+    global changed_options
+    log.d( "notification_callback called with {} options".format( len( opt_list ) ) )
+    for opt in opt_list:
+        log.d( "    Changed option {}".format( opt ) )
+        if not depth_sensor.is_option_read_only( opt ): # Ignore accidental temperature changes
+            changed_options = changed_options + 1
+
+depth_sensor.on_options_changed( notification_callback )
+
+with test.closure( 'disabling auto exposure' ): # Need to disable or changing gain/exposure might automatically disable it
+    depth_sensor.set_option( rs.option.enable_auto_exposure, 0 )
+    test.check_equal( depth_sensor.get_option( rs.option.enable_auto_exposure ), 0.0 )
+    time.sleep( 1.5 ) # default options-watcher update interval is 1 second
+
+with test.closure( 'set one option' ):
+    changed_options = 0
+    current_gain = depth_sensor.get_option( rs.option.gain )
+    depth_sensor.set_option( rs.option.gain , current_gain + 1 )
+    test.check_equal( depth_sensor.get_option( rs.option.gain ), current_gain + 1 )
+    time.sleep( 1.5 ) # default options-watcher update interval is 1 second
+    test.check_equal( changed_options, 1 )
```

Comment:
This makes a pretty big assumption that no other (non-read-only) options will get changed.
How about we test this? Add a closure that makes sure that, in say 3 seconds, no options are changed.

Reply:
Why would options change? We are in the context of a unit test, checking the depth sensor, what option might change?

Reply:
I don't know (and we're not expecting any to), but I don't like assumptions.
Let's test it.

Reply:
Added